addons/mail/static/src/component_hooks/use_messaging_features/use_messaging_features.js

```diff
@@ -0,0 +1,31 @@
+/** @odoo-module */
+
+import { useModels } from "@mail/component_hooks/use_models/use_models";
+import { useShouldUpdateBasedOnProps } from "@mail/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props";
+
+const componentRegistry = {};
+
+/**
+ * Allows a component to lean on the messaging features: the component is
+ * subscribed to changes in messaging models in its environment, the component
+ * will only be rerenderd if its props change, and the component has access to
+ * all components that are registered with this function without having to
+ * explicitly import them.
+ *
+ * @param {function} ComponentClass the constructor of the component to be
+ *      registered. Its name will be used as its key in the registry.
+ * @param {Object} [propsCompareDepth={}] @see useShouldUpdateBasedOnProps
+ */
+export function registerMessagingComponent(ComponentClass, propsCompareDepth = {}) {
```

Comment:
I would have `propsCompareDepth` as a key of a more generic param, because we'll definitely have more possibilities very soon that will require other params now that this PR is opening the way for them.